torch/_higher_order_ops/cond.py

```diff
@@ -28,11 +30,51 @@
 )
 
 
+@contextmanager
+def _set_compilation_env():
+    _old_is_tracing = torch.fx._symbolic_trace._is_fx_tracing_flag
+    _old_capture_dyn_ops = torch._dynamo.config.capture_dynamic_output_shape_ops
+    _old_capture_scalar = torch._dynamo.config.capture_scalar_outputs
+    _old_size_limit = torch._dynamo.config.cache_size_limit
+    _old_acc_size_limit = torch._dynamo.config.accumulated_cache_size_limit
+
+    try:
+        # We need to turn off the is_fx_tracing_flag temporarily
+        # Remove it once we are confident fx tracing works with dynamo
+        torch.fx._symbolic_trace._is_fx_tracing_flag = False
+        torch._dynamo.config.capture_dynamic_output_shape_ops = True
+        torch._dynamo.config.capture_scalar_outputs = True
+        # We need to enlarge the dynamo cache size limit because
+        # cond's frame is expected to be hitted with different inputs frequently
+        torch._dynamo.config.cache_size_limit = 256
+        torch._dynamo.config.accumulated_cache_size_limit = 256
+        yield
+    finally:
+        torch.fx._symbolic_trace._is_fx_tracing_flag = _old_is_tracing
+        torch._dynamo.config.capture_dynamic_output_shape_ops = _old_capture_dyn_ops
+        torch._dynamo.config.capture_scalar_outputs = _old_capture_scalar
+        torch._dynamo.config.cache_size_limit = _old_size_limit
+        torch._dynamo.config.accumulated_cache_size_limit = _old_acc_size_limit
+
+
 @dataclass
 class UnsupportedAliasMutationException(RuntimeError):
     reason: str
 
 
+def cond_compiled(pred, true_fn, false_fn, args):
+    if torch._dynamo.is_compiling() or not torch._dynamo.is_dynamo_supported():
+        return cond(pred, true_fn, false_fn, args)
```

Comment:
Yeah, that's a good point. It's not clear to me how to enforce it. In that case, should we just catch all the unimplemented error sin dynamo for cond and turn them into a different error that won't trigger dynamo re-compiliation so that dynamo won't fall back to eager?